lxd/container_patch.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/lxc/lxd/shared"
+)
+
+func containerPatch(d *Daemon, r *http.Request) Response {
+	// Get the container
+	name := mux.Vars(r)["name"]
+	c, err := containerLoadByName(d, name)
+	if err != nil {
+		return NotFound
+	}
+
+	// Validate the ETag
+	etag := []interface{}{c.Architecture(), c.LocalConfig(), c.LocalDevices(), c.IsEphemeral(), c.Profiles()}
+	err = etagCheck(r, etag)
+	if err != nil {
+		return PreconditionFailed(err)
+	}
+
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		return InternalError(err)
+	}
+
+	rdr1 := ioutil.NopCloser(bytes.NewBuffer(body))
+	rdr2 := ioutil.NopCloser(bytes.NewBuffer(body))
+
+	reqRaw := shared.Jmap{}
+	if err := json.NewDecoder(rdr1).Decode(&reqRaw); err != nil {
+		return BadRequest(err)
+	}
+
+	req := containerPutReq{}
+	if err := json.NewDecoder(rdr2).Decode(&req); err != nil {
+		return BadRequest(err)
+	}
+
+	if req.Restore != "" {
+		return BadRequest(fmt.Errorf("Can't call PATCH in restore mode."))
+	}
+
+	// Check if architecture was passed
+	var architecture int
```

Comment:
Can someone explain (or point me at documentation) how the architecture of a container can be updated without everything exploding?  I'm trying, and failing, to wrap my head around this concept.